test/battle/ability/clear_body.c

```diff
@@ -26,13 +28,241 @@ SINGLE_BATTLE_TEST("Clear Body prevents intimidate")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Speed(4); }
+        OPPONENT(SPECIES_BELDUM) { Speed(6); Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_THUNDER_WAVE); }
+        TURN { MOVE(player, MOVE_THUNDER_WAVE); }
+    } SCENE {
+        NOT ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Foe Beldum used Celebrate!");
+        MESSAGE("Wobbuffet used Thunder Wave!");
+        MESSAGE("Wobbuffet used Thunder Wave!");
+        ONE_OF {
+            MESSAGE("Foe Beldum used Celebrate!");
+            MESSAGE("Foe Beldum is paralyzed! It can't move!");
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent Attack reduction from burn", s16 damage)
+{
+    bool32 burned;
+    PARAMETRIZE{ burned = FALSE; }
+    PARAMETRIZE{ burned = TRUE; }
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_TACKLE].category == DAMAGE_CATEGORY_PHYSICAL);
+        PLAYER(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); if (burned) Status1(STATUS1_BURN); }
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_TACKLE); }
+    } SCENE {
+        NOT ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        HP_BAR(player, captureDamage: &results[i].damage);
+    } FINALLY {
+        EXPECT_MUL_EQ(results[0].damage, Q_4_12(0.5), results[1].damage);
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent receiving negative stat changes from Baton Pass")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SCARY_FACE].effect == EFFECT_SPEED_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_BATON_PASS].effect == EFFECT_BATON_PASS);
+        PLAYER(SPECIES_WOBBUFFET) { Speed(4); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(3); }
+        OPPONENT(SPECIES_BELDUM) { Speed(6); Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_SCARY_FACE); MOVE(opponent, MOVE_BATON_PASS); SEND_OUT(opponent, 1); }
+        TURN { MOVE(player, MOVE_SCARY_FACE); }
+    } SCENE {
+        NOT ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Wobbuffet used Scary Face!");
+        MESSAGE("Foe Beldum used Celebrate!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent Topsy-Turvy")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_TOPSY_TURVY].effect == EFFECT_TOPSY_TURVY);
+        ASSUME(gMovesInfo[MOVE_SCARY_FACE].effect == EFFECT_SPEED_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_BATON_PASS].effect == EFFECT_BATON_PASS);
+        PLAYER(SPECIES_WOBBUFFET) { Speed(4); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(3); }
+        OPPONENT(SPECIES_BELDUM) { Speed(6); Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_SCARY_FACE); MOVE(opponent, MOVE_BATON_PASS); SEND_OUT(opponent, 1); }
+        TURN { MOVE(player, MOVE_TOPSY_TURVY); }
+        TURN { MOVE(player, MOVE_SCARY_FACE); }
+    } SCENE {
+        NOT ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Wobbuffet used Topsy-Turvy!");
```

Comment:
The message doesn't prove if a move was used or not. It needs to check the animation as well. Same principle with the ability check 
move anim playes:
`ANIMATION(ANIM_TYPE_MOVE, MOVE_EMBER, player);`
does not 
`ANIMATION(ANIM_TYPE_MOVE, MOVE_EMBER, player);`

Also the not ability check should be in a places where it technically could go off.

Reply:
Pushed this addition, let me know if I've missed places you'd have liked it added :)